.github/workflows/pythonapp.yml

```diff
@@ -33,7 +33,7 @@ jobs:
         pip install .[tests]
     - name: Run basic tests without extra
       run: |
-        pytest tests/test_no_extra_install.py
+        pytest
```

Comment:
uhhh how is this working?  These should fail because the rest of the libraries aren't installed

Reply:
I added conditions on all tests. I think it is better since we cannot expect users to know which test to run or not run based on their dependencies.

Reply:
Yes, agreed.